src/Plutus/Types/AssocMap.purs

```diff
@@ -200,7 +225,7 @@ mapWithKey
    . (k -> a -> b)
   -> Map k a
   -> Map k b
-mapWithKey f (Map xs) = Map $ map (\(k /\ v) -> k /\ f k v) xs
+mapWithKey = mapWithIndex
```

Comment:
We should remove `mapWithKey` because `mapWithIndex` is the idiomatic way.

Reply:
Ok